corehq/apps/domain/forms.py

```diff
@@ -1331,7 +1331,22 @@ class HQSetPasswordForm(SetPasswordForm):
                                     """))
 
     def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+        from corehq.apps.hqwebapp.utils import decode_password
+        password1 = decode_password(self.cleaned_data.get('new_password1'))
+        return clean_password(password1)
+
+    def clean_new_password2(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        password1 = self.cleaned_data.get('new_password1')
+        password2 = decode_password(self.cleaned_data.get('new_password2'))
+        if password1 and password2:
```

Comment:
Currently its breaking in case decode_password returns an empty string. This can only happen if the decoded password is not properly generated. Also replay attack is just for login.
Fixed by checking for empty password and raised ValidationError with `password_empty` message.